tensorflow/c/experimental/filesystem/plugins/hadoop/hadoop_filesystem.cc

```diff
@@ -169,13 +170,15 @@ class LibHDFS {
   void* handle_;
 };
 
-// We rely on HDFS connection caching here. The HDFS client calls
-// org.apache.hadoop.fs.FileSystem.get(), which caches the connection
-// internally.
+static std::map<std::string, hdfsFS> connectionCache_;
```

Comment:
I think we will have to make some changes here. We have to put it inside a `typedef struct HadoopFile`. Something like this
```cpp
namespace tf_hadoop_filesystem { 
typedef struct HadoopFile {
  LibHDFS* libhdfs;
  std::map<std::string, hdfsFS> connection_cache;
} HadoopFile;
```

And the `Init` will become:
```cpp
  filesystem->plugin_filesystem = new HadoopFile({new LibHDFS(status), {}});
  if (TF_GetCode(status) != TF_OK) return;
  TF_SetStatus(status, TF_OK, "");
```
We will have to modify a lot of code to reflect the change. This
```cpp
auto libhdfs = static_cast<LibHDFS*>(filesystem->plugin_filesystem);
auto fs = Connect(libhdfs, path, status);
```
will become
```cpp
auto hadoop_file = static_cast<HadoopFile*>(filesystem->plugin_filesystem);
auto libhdfs = hadoop_file->libhdfs;
auto fs = Connect(hadoop_file, path, status);
///...
Connect(HadoopFile* hadoop_file, const std::string& path, TF_Status* status ) {
/// ...
  if (hadoop_file->connection_cache.find(cacheKey) == hadoop_file->connection_cache.end()) {
     hadoop_file->connection_cache[cache_key] = libhdfs->hdfsBuilderConnect(builder);
   }
}
```

There are many `HadoopFile` so please pay attentions to the namespace. In addition, I use snake_case here so could you use it here please ? `connectionCache_` is used for private class member which is not the case here.

Thank you.

Reply:
@vnvo2409 
Thanks for your review. I made a modification based on the example you gave. Can you please review it again?